langchain_mcp_adapters/tools.py

```diff
@@ -113,3 +117,57 @@ async def load_mcp_tools(
         for tool in tools.tools
     ]
     return converted_tools
+
+
+def _get_injected_args(tool: BaseTool) -> list[str]:
+    def _is_injected_arg_type(type_: type) -> bool:
+        return any(
+            isinstance(arg, InjectedToolArg)
+            or (isinstance(arg, type) and issubclass(arg, InjectedToolArg))
+            for arg in get_args(type_)[1:]
+        )
+
+    injected_args = [
+        field
+        for field, field_info in get_all_basemodel_annotations(tool.args_schema).items()
+        if _is_injected_arg_type(field_info)
+    ]
+    return injected_args
+
+
+def convert_langchain_tool_to_fastmcp_tool(tool: BaseTool) -> FastMCPTool:
+    """Convert a LangChain tool to a FastMCP tool."""
+    if not issubclass(tool.args_schema, BaseModel):
+        raise ValueError(
+            "Tool args_schema must be a subclass of pydantic.BaseModel. "
+            "Tools with dict args schema are not supported."
+        )
+
+    parameters = tool.tool_call_schema.model_json_schema()
+    field_definitions = {
+        field: (field_info.annotation, field_info)
+        for field, field_info in tool.tool_call_schema.model_fields.items()
+    }
+    arg_model = create_model(
+        f"{tool.name}Arguments",
+        **field_definitions,
+        __base__=ArgModelBase,
+    )
+    fn_metadata = FuncMetadata(arg_model=arg_model)
+
+    async def fn(**arguments: dict[str, Any]) -> Any:
+        return await tool.ainvoke(arguments)
+
+    injected_args = _get_injected_args(tool)
+    if len(injected_args) > 0:
+        raise NotImplementedError("LangChain tools with injected arguments are not supported")
```

Comment:
is runnable config is an injected argument I assume?

I don't think users will know that this is the case since it's not called an injected argument